Objects/unicodeobject.c

```diff
@@ -8964,21 +8964,19 @@ _PyUnicode_InsertThousandsGrouping(
     return count;
 }
 
-
-Py_ssize_t
-PyUnicode_Count(PyObject *str,
-                PyObject *substr,
-                Py_ssize_t start,
-                Py_ssize_t end)
+static Py_ssize_t
+any_unicode_count(PyObject *str,
+                  PyObject *substr,
+                  Py_ssize_t start,
+                  Py_ssize_t end)
 {
+    // You must ensure that `str` and `substr` are both unicode objects
```

Comment:
Instead of comment use `assert(PyUnicode_CheckExact(foo))` .

Reply:
Thanks!

Reply:
I went with `assert(PyUnicode_Check(str))`, because subclasseses of `str` must work as well.